nilearn/input_data/nifti_labels_masker.py

```diff
@@ -440,11 +441,25 @@ def transform_single_imgs(self, imgs, confounds=None, sample_mask=None):
             if not _check_same_fov(imgs_, self._resampled_labels_img_):
                 if self.verbose > 0:
                     print("Resampling labels")
+                labels_before_resampling = set(
+                    np.unique(_safe_get_data(self._resampled_labels_img_))
+                )
                 self._resampled_labels_img_ = self._cache(
                     image.resample_img, func_memory_level=2)(
                         self.labels_img_, interpolation="nearest",
                         target_shape=imgs_.shape[:3],
                         target_affine=imgs_.affine)
+                labels_after_resampling = set(
+                    np.unique(_safe_get_data(self._resampled_labels_img_))
+                )
+                labels_diff = labels_before_resampling - labels_after_resampling
+                if len(labels_diff) > 0:
+                    warnings.warn("After resampling the label image to the "
```

Comment:
@jeromedockes Yes it shows the label indices. I initially wanted to show the actual labels, which are stored in `self.labels` (if provided by the user), but I am unhappy by the way the mapping is done atm. 
Maintaining proper indexing of multiple lists which are created at different times, and which can change in length depending on the transform inputs, or the fact that the user did or nor provide a list of labels with "background" in it is a pain. Even writing basic length checks becomes tricky.
I'm starting to think that we should have asked for an actual mapping (i.e a <label_int> : <label_str> dictionary) instead of a simple list of labels.
However, changing this parameter to a dict would require a deprecation cycle and I'm not sure if it is worth it atm.... :thinking: 
Does this make sense?

Reply:
thanks, that makes sense. I agree with you about the fact a mapping rather than list indices might have been better, and about the fact that changing it now is not a good idea. we can keep the warning as it is then!